test/util/regtest.js

```diff
@@ -0,0 +1,321 @@
+'use strict';
+
+const {NodeClient, WalletClient} = require('bclient');
+
+const assert = require('bsert');
+const {sleep} = require('./common');
+const FullNode = require('../../lib/node/fullnode');
+const SPVNode = require('../../lib/node/spvnode');
+const Coin = require('../../lib/primitives/coin');
+const MTX = require('../../lib/primitives/mtx');
+
+async function initFullNode(options) {
+  const node = new FullNode({
```

Comment:
This applies to all init methods down here.

The configs are too specific, so maybe leave defaults here as they are but add possibility to override them with `options`.

```
... = new FullNode({
   ...
   logLevel: options.logLevel,
   ...options
});
```

or pass another optional parameter, `extendOptions = {}` that can be applied here, to avoid
collisions from custom configuration options (e.g. `options.ports`)

Reply:
This is intended to be used with the pagination tests with the options needed for those cases. This may not end up being useful for other test cases, the other methods for generating transactions and blocks are specific to the test case. I think it's best to leave without the options, until needed.